packages/windows/src/utils.ts

```diff
@@ -0,0 +1,131 @@
+import { App } from '@navaiguide/core';
+import { exec } from 'child_process';
+import path from 'path';
+
+
+function runPowerShellModuleFunction(functionName: string, namedArgs: {[key: string]: string | number} = {}): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const modulePath = path.join(__dirname, 'WinAutomation.psm1');
+    
+    // Construct the command arguments string with named parameters
+    const commandArgs = Object.entries(namedArgs)
+                              .map(([key, value]) => `-${key} "${value.toString().replace(/ /g, '%20').replace(/"/g, '\\"')}"`)
+                              .join(' ');
+    
+    const command = `powershell -Command "& {Import-Module -Name '${modulePath}'; ${functionName} ${commandArgs}}"`;
+
+    exec(command, { maxBuffer: 1024 * 1024 * 5 }, (error, stdout, stderr) => {
+      if (error) {
+        reject(error.message);
+      } else if (stderr) {
+        reject(stderr);
+      } else {
+        resolve(stdout.trim());
+      }
+    });
+  });
```

Comment:
Suggestion: by using Node's [`promisify`](https://nodejs.org/api/util.html#utilpromisifyoriginal) on `exec` this can be turned into a bit nicer `await` syntax.